scripts/start/readCheckScenarioConfig.R

```diff
@@ -0,0 +1,86 @@
+#' read a REMIND scenario_config*.csv file, make sure it contains all columns specified
+#' in path_gdx_list.
+#' Checks whether scenario titles are neither too long, don't contain dots and don't end with a _
+#' Checks for outdated and unknown column titles
+#'
+#' @param filename string with scenario_config*.csv filename
+#' @param remindPath path to remind main directory
+#' @param testmode if TRUE, generates warnings if unknownColumnNames exist
+#' @author Oliver Richters
+#' @return list with scenario config content
+readCheckScenarioConfig <- function(filename, remindPath = ".", testmode = FALSE) {
+  if (testmode) {
+    cfg <- suppressWarnings(gms::readDefaultConfig(remindPath))
+  } else {
+    cfg <- gms::readDefaultConfig(remindPath)
+  }
+  scenConf <- read.csv2(filename, stringsAsFactors = FALSE, row.names = 1, na.strings = "", comment.char = "#")
+  toolong <- nchar(rownames(scenConf)) > 75
+  if (any(toolong)) {
+    warning("These titles are too long: ",
+            paste0(rownames(scenConf)[toolong], collapse = ", "),
+            " – GAMS would not tolerate this, and quit working at a point where you least expect it. Stopping now.")
+  }
+  regionname <- grepl("^(([A-Z]{3})|(glob))$", rownames(scenConf))
+  if (any(regionname)) {
+    warning("These titles may be confused with regions: ",
+            paste0(rownames(scenConf)[regionname], collapse = ", "),
+            " – Titles with three capital letters or 'glob' may be confused with region names by magclass. Stopping now.")
+  }
+  containsdot <- grep("\\.", rownames(scenConf))
+  if (any(containsdot)) {
+    warning("These titles contain a dot: ",
+            paste0(rownames(scenConf)[containsdot], collapse = ", "),
+            " – GAMS would not tolerate this, and quit working at a point where you least expect it. Stopping now.")
+  }
+  underscore <- grep("_$", rownames(scenConf))
+  if (any(underscore)) {
+    warning("These titles end with _: ",
+            paste0(rownames(scenConf)[underscore], collapse = ", "),
+            ". This may lead to wrong gdx files being selected. Please use # to make comments, not underscores.",
+            " Stopping now.")
+  }
+  if ("path_gdx_ref" %in% names(scenConf) && ! "path_gdx_refpolicycost" %in% names(scenConf)) {
+    scenConf$path_gdx_refpolicycost <- scenConf$path_gdx_ref
+    message("In ", filename,
+        ", no column path_gdx_refpolicycost for policy cost comparison found, using path_gdx_ref instead.")
+  }
+  errorsfound <- sum(toolong) + sum(regionname) + sum(containsdot) + sum(underscore)
+  scenConf[, names(path_gdx_list)[! names(path_gdx_list) %in% names(scenConf)]] <- NA
```

Comment:
`path_gdx_list` is accessed without being supplied explicitly. However, in the new structure it is sourced together with the scripts if I am not mistaken. So we use it like a global constant now?

Reply:
In a way, yes. But is true that you cannot source only this file here and run `readCheckScenarioConfig`, without also sourcing `path_gdx_list`. Not sure whether that is a problem.

Reply:
`readCheckScenarioConfig` seems to be a quite specific function . Thus, I would suggest to keep it as is and change it if we observe that `readCheckScenarioConfig` is used elsewhere.

Reply:
A `if (! exists("path_gdx_list")) stop("You have to source path_gdx_list.R as well!")` wouldn't cost us much, on the other hand. Same for `configureCfg`.